### ISSUE TITLE
cf2 file fields, only one file is attached and uploaded

### ISSUE THREAD

**Issue by Shelob9** [CalderaWP/Caldera-Forms#2818]
@New0  found that for files field, when multiple is enabled, only one file is  attached to mailer and one file uploaded to media library.

**Comment by Shelob9** [comment]
Relateded sort of: #2814

**Comment by Shelob9** [comment]
1) There is a typo in the inline docs about what type of argument for function `hashAndUpload` in render/index.js

2) I observed that if I have one file field and I add two files, when I submit, I only see 2 HTTP requests, I would expect three. In the media library I see that only the first was added, I would expect to see two.

The tests here, which are server-side tests prove that if the right data was already on the server, it would attach both.
* https://github.com/CalderaWP/Caldera-Forms/blob/develop/tests/Integration/SubmissionsWithFileFieldsTest.php
* https://github.com/CalderaWP/Caldera-Forms/blob/develop/tests/Integration/FileUploadTest.php

I think the problem is purely in the client, only one upload happens before allowing submission to complete.

**Comment by Shelob9** [comment]
After uploading first file ~L155 clients/render/index.js  the field is removed from pending, without triggering upload of second file.
```
else if (response.hasOwnProperty('control')) {
									removeFromPending(fieldId);
									removeFromBlocking(fieldId);
									cf2.uploadCompleted.push(fieldId);
									$form.submit();
								}else{```

**Comment by Shelob9** [comment]
To make this testable, we need to move the closure for `cf.ajax.request` a function. Then we need to make the response handlers separate functions. Then we can test that if there are 2 files, only gets removed from cf2.pending if both are uploaded. We need to test that the cf2 object is updated correctly. 

Given that file field has 2 files. Expect one file uploaded and submit is still blocked based on that field. Given that second file has uploaded, expect submission not to be blocked.

Mocking jQuery - https://stackoverflow.com/a/45997857/1469799

Pseudo-tests https://gist.github.com/Shelob9/0ed44838d53e3aacd349136f5a6d7135#file-pseudo-tests-js

**Comment by New0** [comment]
Fixed by #2910
